### PR TITLE
[ORCA-249] Push docker image to ghcr and use trivy docker scan tool

### PR DESCRIPTION
**Pull request by rxu17**
**Purpose:** This PR pushes the docker image to GHCR and also uses [trivy-action](https://github.com/aquasecurity/trivy-action) to scan the docker images whenever changes are made to `dsl2` branch, and also scans on a weekly basis. Please ignore the millions of commits, will squash them when this is approved to merge.

You can see a sample output of high/critical scans from trivy under the Security Tab (here on github) -> Code scanning -> Search for the branch orca-249-push-to-ghcr. 

Note that when we eventually merge dsl2 -> main, the branches on the workflows will have to be updated to `main`

### CODE REVIEW

**Review by thomasyu888 — APPROVED** [review]
I'm going to pre-approve, but I asked @adamjtaylor if we can just merge dsl2 branch into main so we can just go ahead and make those changes

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -3,43 +3,55 @@ name: ci
 on:
   push:
     branches:
-      - 'main'
-      - 'dsl2'
+     - 'dsl2'
     paths:
      - 'docker/**'
+     - '.github/workflows/ci.yml'
```
[the remark this answers is not in the document]
Comment (rxu17) [reply .github/workflows/ci.yml:9]: It determines when to run the GH actions to push the image to ghcr -- **only** when there are changes to any of the files under `docker/**` and when there are changes to the ci workflow file itself made to `dsl2` branch.

[the remark this answers is not in the document]
Comment (thomasyu888) [reply .github/workflows/ci.yml:9]: TIL: this....

**Review by adamjtaylor — APPROVED** [review]

**Comment by rxu17** [comment]
Some fun behavior in the Dockerfile was that it was defaulting to installing `tiffslide 2.2.0` which requires `zarr>=2.11.0`, but we have `zarr 2.6.1` which is incompatible. Even though zarr is mentioned in our `environment.yml` file, and conda does initially install a zarr version that is compatible with `tiffslide 2.2.0`, there is a line in the Dockerfile: `pip install git+https://github.com/labsyspharm/minerva-lib-python@master#egg=minerva-lib` that installs this minerva python library which sets zarr version to `zarr 2.6.1` so it overrides what conda installed before. And since pip doesn't track versions installed by conda and itself for compatibility issues, `tiffslide 2.2.0` and `zarr 2.6.1` were both allowed to exist causing the following issue when running nf-artist.

```Command error:
  Traceback (most recent call last):
    File ".command.sh", line 3, in <module>
      from tiffslide import TiffSlide
    File "/venv/lib/python3.10/site-packages/tiffslide/__init__.py", line 9, in <module>
      from tiffslide.tiffslide import PROPERTY_NAME_BACKGROUND_COLOR
    File "/venv/lib/python3.10/site-packages/tiffslide/tiffslide.py", line 40, in <module>
      from tiffslide._zarr import get_zarr_depth_and_dtype
    File "/venv/lib/python3.10/site-packages/tiffslide/_zarr.py", line 17, in <module>
      from zarr.storage import BaseStore
  ImportError: cannot import name 'BaseStore' from 'zarr.storage' (/venv/lib/python3.10/site-packages/zarr/storage.py)
```

This was resolved by checking the version of `tiffslide` in the dockerhub docker file, and enforcing that version in requirements.yml. In the future though, we'll have to keep in mind that if we ever want to update the version of  `tiffslide` we have, we would have to update the `zarr` version in the [minerva python package](https://github.com/labsyspharm/minerva-lib-python) if we want to maintain compatibility.

**Pull request merged**